src/requests/cache/sorted_set/sorted_set_put_elements.rs

```diff
@@ -1,17 +1,104 @@
+use std::collections::HashMap;
+
 use momento_protos::cache_client::{SortedSetElement, SortedSetPutRequest};
 
 use crate::requests::cache::MomentoRequest;
 use crate::simple_cache_client::prep_request_with_timeout;
 use crate::{CacheClient, CollectionTtl, IntoBytes, MomentoResult};
 
+/// This trait defines an interface for converting a type into a vector of `SortedSetElement`s.
+pub trait IntoSortedSetElements: Send {
```

Comment:
hmm, TIL; I was wondering why we are doing stuff this way for, e.g., IntoBytes; like, why aren't we just doing `impl From<Vec<(V, f64)>> for Vec<SortedSetElement>`?

Then I tried to impl that and it looks like the reason is that rust won't let you define an impl for a trait if both the trait and the struct you are trying to impl for are not defined in the current crate. That's a little bit of a bummer. Would be nice to just be able to do `.into()`, but, not a huge deal.

Reply:
Agreed. I tried a few alternatives thinking about how much boilerplate the interface introduces. Eg what we really want is for the trait to apply if your type has an iterator over tuples `(IntoBytes, f64)`. Then our trait would have a default method to convert from such an iterator to `Vec<SortedSetPutElement>`.